base/reflection.jl

```diff
@@ -221,6 +228,12 @@ function code_typed(@nospecialize(f), @nospecialize(types=default_tt(f)); kwargs
     return code_typed_by_type(tt; kwargs...)
 end
 
+# support 'functor'-like queries, such as `(::Foo)(::Int, ::Int)` via `code_typed((Foo, Int, Int))`
+function code_typed(@nospecialize(types::Union{Tuple,Array,SimpleVector}); kwargs...)
```

Comment:
Is there any reason to not only include `types::Tuple`? The limitation on shadowing `(::Tuple)(...)` would apply to `SimpleVector` and `Array`, which seems a bit more impactful (particularly for `Array`).